lib/Twig/Environment.php

```diff
@@ -314,8 +325,7 @@ public function getTemplateClass($name, $index = null)
     {
         $key = $this->getLoader()->getCacheKey($name);
         $key .= json_encode(array_keys($this->extensions));
```

Comment:
Why not moving extensions as well in the hash? That would avoid the `json_encode` call.